sources/en/0/888novel.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+
+import logging
+from lncrawl.core.crawler import Crawler
+from lncrawl.utils.cleaner import TextCleaner
+from bs4 import Comment, Tag
+
+logger = logging.getLogger(__name__)
+
+
+class Eight88NovelCrawler(Crawler):
+    base_url = ["https://888novel.com/"]
+
+    has_manga = False
+    machine_translation = False
+
+    def initialize(self) -> None:
+        class Eight88NovelTextCleaner(TextCleaner):
+            """A copy of TextCleaner that keep double <br> tags"""
+
+            def clean_contents(self, div):
+                if not isinstance(div, Tag):
+                    return div
+                # end if
+                if self.bad_css:
+                    for bad in div.select(",".join(self.bad_css)):
+                        bad.extract()
+                    # end if
+                # end if
+                for tag in div.find_all(True):
+                    if isinstance(tag, Comment):
+                        tag.extract()  # Remove comments
+
+                    # I want to keep double <br> tags
```

Comment:
Why do you want to keep <br> tags? In some epub readers it makes the gap between two paragraphs too wide for comfort.

Reply:
For me it looks like a single block with one ```<br>``` using sumatrapdf on windows or moon+reader on android. 

![double br](https://user-images.githubusercontent.com/86294972/172021975-99317145-4112-4c44-b448-bd5c08d914c5.png)

![single br](https://user-images.githubusercontent.com/86294972/172021977-f163358c-eef0-41f3-abb7-f84289a7c22f.png)

Reply:
ah, I get it. i think i should remove the initialize funtion.

you can use calibre for reading. it has some nice features for formatting ebooks for reading.